spouse/spouse_demo.py

```diff
@@ -0,0 +1,552 @@
+#
+# There are a few advantages of having preprocessors and labeling functions in this form:
+#
+# **Data Agnostic:**  Operate over multiple data types without rewriting
+#
+# **Incremental Processing:** Can create preprocessors as needed while writing LFs!
+#
+# **Future Use:** Can store them for later for different tasks since they are reproducible and modular
+#
+# **Optimizations:** Allows caching behind-the-scenes
+
+# %%
+from snorkel.labeling.apply import PandasLFApplier
+from snorkel.labeling.lf import labeling_function
+
+from preprocessors import get_left_tokens, get_person_last_names
+
+POSITIVE = 1
+NEGATIVE = 0
+ABSTAIN = -1
+
+# %%
+# Check for the `spouse` words appearing between the person mentions
+spouses = {"spouse", "wife", "husband", "ex-wife", "ex-husband"}
+
+
+@labeling_function(resources=dict(spouses=spouses))
+def lf_husband_wife(x, spouses):
+    return POSITIVE if len(spouses.intersection(set(x.between_tokens))) > 0 else ABSTAIN
+
+
+# %%
+# Check for the `spouse` words appearing to the left of the person mentions
+@labeling_function(resources=dict(spouses=spouses), preprocessors=[get_left_tokens])
+def lf_husband_wife_left_window(x, spouses):
+    if len(set(spouses).intersection(set(x.person1_left_tokens))) > 0:
+        return POSITIVE
+    elif len(set(spouses).intersection(set(x.person2_left_tokens))) > 0:
+        return POSITIVE
+    else:
+        return ABSTAIN
+
+
+# %%
+# Check for the person mentions having the same last name
+@labeling_function(preprocessors=[get_person_last_names])
+def lf_same_last_name(x):
+    p1_ln, p2_ln = x.person_lastnames
+
+    if p1_ln and p2_ln and p1_ln == p2_ln:
+        return POSITIVE
+    return ABSTAIN
+
+
+# %%
+# Check for the word `married` between person mentions
+@labeling_function()
+def lf_married(x):
+    return POSITIVE if "married" in x.between_tokens else ABSTAIN
+
+
+# %%
+# Check for words that refer to `family` relationships between and to the left of the person mentions
+family = [
+    "father",
+    "mother",
+    "sister",
+    "brother",
+    "son",
+    "daughter",
+    "grandfather",
+    "grandmother",
+    "uncle",
+    "aunt",
+    "cousin",
+]
+family = set(family + [f + "-in-law" for f in family])
+
+
+@labeling_function(resources=dict(family=family))
+def lf_familial_relationship(x, family):
+    return NEGATIVE if len(family.intersection(set(x.between_tokens))) > 0 else ABSTAIN
+
+
+@labeling_function(resources=dict(family=family), preprocessors=[get_left_tokens])
+def lf_family_left_window(x, family):
+    if len(set(family).intersection(set(x.person1_left_tokens))) > 0:
+        return NEGATIVE
+    elif len(set(family).intersection(set(x.person2_left_tokens))) > 0:
+        return NEGATIVE
+    else:
+        return ABSTAIN
+
+
+# %%
+# Check for `other` relationship words between person mentions
+other = {"boyfriend", "girlfriend", "boss", "employee", "secretary", "co-worker"}
```

Comment:
change from set to list to be consistent with above examples

Reply:
We need sets to compute intersection. So I changed the above (family) to be a set instead.